src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -607,6 +609,9 @@ public static IResourceBuilder<T> WaitFor<T>(this IResourceBuilder<T> builder, I
                     $"Resource '{dependency.Resource.Name}' has entered the '{snapshot.State.Text}' state prematurely."
                     );
             }
+
+            // If we get to here the resource is running so we just need to check on whether it is healthy or not.
+            await rhs.WaitUntilResourceHealthyAsync(dependency.Resource, ct).ConfigureAwait(false);
```

Comment:
This shouldn't exist, instead we should tweak the predicate to include the health state of the resource.

Reply:
Yeah I removed it.